### PR TITLE
Update DG, Tag , findcourse, findstudent, use cases, plan enhancement

### PR DESCRIPTION
**Pull request by tanteckfang**

### CODE REVIEW

File: docs/images/TagSequenceDiagram.png
Comment (joel-foo) [inline docs/images/TagSequenceDiagram.png:1]: Hi, for this, we should not show the call setTag from Tag to TagType, as this is internal to Model component.

**Review by joel-foo — CHANGES_REQUESTED** [review]
LGTM, except for a slight change.

File: docs/images/TagSequenceDiagram.png
Comment (tanteckfang) [reply docs/images/TagSequenceDiagram.png:1]: Hi Joel, I have made some changes and I would like to focus on Tag, I had combined the Logic into one participant.

**Review by joel-foo — APPROVED** [review]
LGTM!

**Pull request merged**